lib/PlatoonService/src/Waypoint.cpp

```diff
@@ -0,0 +1,151 @@
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/** Default size of the JSON Document for parsing. */
+static const uint32_t JSON_DOC_DEFAULT_SIZE = 1024U;
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+Waypoint* Waypoint::deserialize(const String& serializedWaypoint)
+{
+    Waypoint*                                 waypoint = nullptr;
+    StaticJsonDocument<JSON_DOC_DEFAULT_SIZE> jsonPayload;
+    DeserializationError                      error = deserializeJson(jsonPayload, serializedWaypoint.c_str());
+
+    if (error != DeserializationError::Ok)
+    {
+        LOG_ERROR("JSON Deserialization Error %d.", error);
+    }
+    else
+    {
+        JsonVariant jsonXPos        = jsonPayload["X"];           /**< X position [mm]. */
+        JsonVariant jsonYPos        = jsonPayload["Y"];           /**< Y position [mm]. */
+        JsonVariant jsonOrientation = jsonPayload["Orientation"]; /**< Orientation [mrad]. */
+        JsonVariant jsonLeft        = jsonPayload["Left"];        /**< Left motor speed [steps/s]. */
+        JsonVariant jsonRight       = jsonPayload["Right"];       /**< Right motor speed [steps/s]. */
+        JsonVariant jsonCenter      = jsonPayload["Center"];      /**< Center speed [steps/s]. */
+
+        if ((false == jsonXPos.isNull()) && (false == jsonYPos.isNull()) && (false == jsonOrientation.isNull()) &&
+            (false == jsonLeft.isNull()) && (false == jsonRight.isNull()) && (false == jsonCenter.isNull()))
+        {
+
+            int32_t xPos        = jsonXPos.as<int32_t>();
+            int32_t yPos        = jsonYPos.as<int32_t>();
+            int32_t orientation = jsonOrientation.as<int32_t>();
+            int16_t left        = jsonLeft.as<int16_t>();
+            int16_t right       = jsonRight.as<int16_t>();
+            int16_t center      = jsonCenter.as<int16_t>();
+
+            waypoint = new (std::nothrow) Waypoint(xPos, yPos, orientation, left, right, center);
+        }
+    }
+
+    return waypoint;
+}
+
+String Waypoint::serialize() const

```

Comment:
By using a reference to a String, you can avoid that temporary the serialized waypoint will be twice on the heap and stack.